torch_geometric/nn/aggr/attention.py

```diff
@@ -53,28 +54,21 @@ def reset_parameters(self):
         reset(self.gate_nn)
         reset(self.nn)
 
-    def forward(self, x: Tensor, batch: Optional[Tensor] = None,
-                size: Optional[int] = None) -> Tensor:
-        r"""
-        Args:
-            x (Tensor): The input node features.
-            batch (LongTensor, optional): A vector that maps each node to its
-                respective graph identifier. (default: :obj:`None`)
-            size (int, optional): The number of graphs in the batch.
-                (default: :obj:`None`)
-        """
-        if batch is None:
-            batch = x.new_zeros(x.size(0), dtype=torch.int64)
+    def forward(self, x: Tensor, index: Optional[Tensor] = None,
+                ptr: Optional[Tensor] = None, dim_size: Optional[int] = None,
+                dim: int = -2) -> Tensor:
+
```

Comment:
We can do this, but right now I think it would be backwards compatibility breaking because of
```
        x = x.unsqueeze(-1) if x.dim() == 1 else x
```
WDYT?

Reply:
I've changed for now and removed this line. But still interested in your thought here.

Reply:
I personally think this is okay. The change on user-side would be quick, and I assume this case is very rare.